geode-core/src/main/java/org/apache/geode/cache/query/internal/DefaultQueryService.java

```diff
@@ -403,9 +403,12 @@ public Collection getIndexes(Region region) {
     if (region instanceof PartitionedRegion) {
       return ((PartitionedRegion) region).getIndexes();
     }
+
     IndexManager indexManager = IndexUtils.getIndexManager(cache, region, false);
-    if (indexManager == null)
-      return null;
+    if (indexManager == null) {
+      return Collections.emptyList();
+    }
+
```

Comment:
There is also the getIndexes(Region region, IndexType indexType) that should be modified like this as well.

Reply:
Hey @boglesby ,

I saw that but didn't make the change because the method is deprecated and shouldn't be used, according to the interface's [documentation](https://github.com/apache/geode/blob/527e346334b1fc35e6cb8813ed3a48f58a5d11c4/geode-core/src/main/java/org/apache/geode/cache/query/QueryService.java#L486).

```
@deprecated As of 6.6.2, use {@link #getIndexes(Region)} only.
```

Will make the change anyway, it's only one line so no effort at all.
Cheers.